### ISSUE TITLE
can not plot embedded data points using tensorboard embedding chart

### ISSUE THREAD

**Issue by robbine** [tensorflow/tensorflow#7006]
I tried to plot original input data points and embedded data points using tensorboard embedding function.
I did basically everything according to the official instruction.
There are three tensors in my graph, namely input, embedding, and embedded.
and embedded = tf.MatMul(input, embedding)
I failed to plot the embedded tensor, even thought I provide metadata file , tensorboard seems just ignore it.
However I did plot the input tensor, tensorboard has no problem locating metadata file, it shows all the points in 3D with labels I provided in the metadata file.
I also succeeded to plot the embedding tensor, So does anyone know how to solve this?

**Comment by robbine** [comment]
here is a more detailed version, I omitted training code to make it more readable
`        graph = tf.Graph()
        with graph.as_default():
            with tf.name_scope('embeddings'):
                        sp_tensor_map = tf.placeholder(tf.int32, shape=[None, n], name='input')
                        embeddings = tf.Variable(tf.random_normal([n, embedding_size]), name='embedding')
                        dense_embedding = tf.matmul(tf.cast(sp_tensor_map, tf.float32, name='tensor'), embeddings, name='embedded')
            weights = tf.Variable(tf.random_normal([128, 1], stddev=0.01, name='random-normal'), name='weights')
            logits = tf.matmul(dense_embedding, weights, name='logits')
            labels_batch = tf.placeholder(tf.float32, name='y-input')
            cross_entropy = tf.reduce_mean(tf.nn.sigmoid_cross_entropy_with_logits(logits, labels_batch, name='cross_entropy'), name='cross_entropy_mean')
            train_op = tf.train.AdamOptimizer().minimize(cross_entropy)
            tmp_tensor = tf.Variable(valid_single_data)
            result = tf.matmul(tf.cast(tmp_tensor, tf.float32), embeddings, name='res')
        with tf.Session(graph=graph) as sess:
            projector_config = projector.ProjectorConfig()
            add_embedding = projector_config.embeddings.add()
            add_embedding.tensor_name = result.name
            add_embedding.metadata_path = metadata
            summary_writer = tf.summary.FileWriter(log_dir + '/projector', sess.graph)
            projector.visualize_embeddings(summary_writer, projector_config)
            sess.run(tf.global_variables_initializer())
            sess.run(tf.local_variables_initializer())
            import glob
            idx = 0
            valid_feed_dictionary = {
                sp_tensor_map = valid_single_data,
            }
            for filename in glob.glob(path):
                for (single_features_batch_data, combine_features_batch_data, labels_batch_data) in data_iterator(filename):
                        summary, area_under_roc = sess.run([merged, auc], feed_dict= valid_feed_dictionary)
                        test_writer.add_summary(summary, idx)
                        print('step {}: auc is {} '.format(idx, area_under_roc))
                        sess.run(result)
                        saver = tf.train.Saver()
                        saver.save(sess, log_dir + '/model.ckpt', global_step=idx)
            summary_writer.close()`

**Comment by robbine** [comment]
It seems that intermediate results are NOT saved in a checkpoint file

**Comment by asimshankar** [comment]
@robbine : I'm not sure I fully understand your problem, but just to be clear, [`tf.train.Saver`](https://www.tensorflow.org/api_docs/python/state_ops/saving_and_restoring_variables#Saver) saves only variables to the checkpoint - it does not save the output of all nodes in the graph.

Does that answer your question?

**Comment by robbine** [comment]
Yes, I managed to save embedded results by copying it to another variable, 
copy_op = result.assign(embedded)
and it solved my problem. Thanks
